Add support for multiple documents in the input stream

suranc opened the pull request:
This changes Patch.Apply() to use the Decode() method provided by the yaml package instead of Unmarshal(), since Decode() supports decoding multiple documents.

suranc committed:
Switching to Decode from Unmarshal to support multiple documents in the input

suranc committed:
Updating error

suranc committed:
Switching error check after check for last file

suranc committed:
Fixing output by switching to Encode from Marshal

krishicks commented:
Sorry for the suuuuuuper late response to this.

Would you mind adding some tests to show how this would be used?

suranc committed:
Merge remote-tracking branch 'origin/master' into multi-file

suranc committed:
Switch positive test cases to Decode from Unmarshal

suranc committed:
Switch extended syntax test cases to Decode from Unmarshal

suranc committed:
Add failure test cases for Decode change

suranc commented:
I switched the `patch_test.go` file to use Decode instead of Unmarshal to match the `patch.go` code for decoding input.  I also added some positive and negative test cases using multiple documents as input.

krishicks commented:
Cool, thanks!
